src/components/AppNavigationItem/AppNavigationItem.vue

```diff
@@ -158,40 +361,222 @@ export default {
 			this.opened = !this.opened
 			this.$emit('update:open', this.opened)
 		},
-		callPreventStop(action, event) {
-			if (action) {
-				event.preventDefault()
-				event.stopPropagation()
-				action()
-			}
-		},
-		cancelEdit(e) {
+		cancelEdit() {
 			// remove the editing class
-			if (Array.isArray(this.item.classes)) {
-				this.item.classes = this.item.classes.filter(item => item !== 'editing')
-			}
-			this.item.edit.reset(e)
+			this.editing = false
 		},
-		// This is used to decide which outter element type to use
-		// li or router-link
-		navElement(item) {
-			if (item.router) {
-				let exact = item.router.exact
-				if (typeof item.router.exact === 'undefined') {
-					exact = true
-				}
+		onClick(event) {
+			this.$emit('click', event)
+		},
+		handleEdit() {
+			this.editing = true
+		},
+		handleRename() {
+			this.$emit('update:title', this.newTitle)
+			this.newTitle = ''
+			this.editing = false
+		},
+		handleUndo() {
+			this.$emit('undo')
+		}
+	}
+}
+</script>
 
-				return {
-					is: 'router-link',
-					tag: 'li',
-					to: item.router,
-					exact
+<style lang="scss" scoped>
+@import '~Fonts/scss/iconfont-vue';
+
+.app-navigation-entry {
+	position: relative;
+	min-height: $clickable-area;
+	width: 100%;
+	box-sizing: border-box;
+	display: flex;
+	flex-wrap: wrap;
+	padding-right: 6px;
+	order: 1;
+	flex-shrink: 0;
+
+	a:hover,
+	a:focus,
+	a:active {
+		color: var(--color-main-text);
+		box-shadow: inset 4px 0 var(--color-primary);
+	}
+
+	/* hide deletion/collapse of subitems */
+	&.app-navigation-entry--deleted,
+	&.app-navigation-entry--collapsible:not(.app-navigation-entry--opened) {
+		> ul {
+			// NO ANIMATE because if not really hidden, we can still tab through it
+			display: none;
+		}
+	}
+
+	// Main entry link
+	.app-navigation-entry-link {
+		background-size: $icon-size $icon-size;
+		background-position: $icon-margin center;
+		background-repeat: no-repeat;
+		display: flex;
+		line-height: $clickable-area;
+		min-height: $clickable-area;
+		padding: 0;
+		padding-right: $icon-margin;
+		overflow: hidden;
+		box-sizing: border-box;
+		white-space: nowrap;
+		text-overflow: ellipsis;
+		color: var(--color-text-light);
+		flex: 1 1 0px;
+		z-index: 100; /* above the bullet to allow click*/
+
+		.app-navigation-entry-icon {
+			width: $clickable-area;
+			height: $clickable-area;
+			display: flex;
+			flex: 0 0 $clickable-area;
+			justify-content: center;
+			align-items: center;
+		}
+	}
+
+	/* Second level nesting for lists */
+	.app-navigation-entry__children {
```

Comment:
This is the `<ul>` which contains flex elements. Therefore, we need to add
```suggestion
	.app-navigation-entry__children {
		display: flex;
		flex-direction: column;
```

Otherwise, there is a gap of 6px between the children items.